### PR TITLE
Moving function quantity_allclose #7235

### LINKED ISSUE #7235: quantity_allclose can not be imported without pytest installed
```python
>>> from astropy.tests.helper import quantity_allclose
  File "/opt/miniconda/envs/clean/lib/python3.6/site-packages/astropy/tests/helper.py", line 14, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

Is this a recent change? I feel these "all close" helpers should be somewhere else? I am happy to PR this if people agree.

### PR DESCRIPTION
**Pull request by rohanrajpal**
Deprecated the quantity_allclose in astropy.tests.helper
Moved function quantity_allclose to astropy.units.quantity
Closes #7235

### CODE REVIEW

File: astropy/tests/helper.py [hunk 1]
```diff
@@ -465,7 +464,7 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
                                                                rtol, atol),
                                **kwargs)
 
-
+@deprecated("3.1.dev21533")
```
Comment (pllim) [inline astropy/tests/helper.py]: * Just "3.1" is sufficient.
* Please provide a string to explain where it has moved to.
* To avoid code duplication, this function now should only have one functional line of code to just call the new one.

File: astropy/tests/helper.py [hunk 2]
```diff
@@ -3,14 +3,13 @@
 This module provides the tools used to internally run the astropy test suite
 from the installed astropy.  It makes use of the `pytest` testing framework.
 """
-
+from ..utils.decorators import deprecated
```
Comment (pllim) [inline astropy/tests/helper.py]: Please move local import to after `import pytest`.

File: astropy/units/quantity.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 
 # Standard library
+
```
Comment (pllim) [inline astropy/units/quantity.py]: Please remove extra line here.

File: astropy/units/quantity.py [hunk 4]
```diff
@@ -107,7 +117,6 @@ def __next__(self):
 
     next = __next__
 
-
```
Comment (pllim) [inline astropy/units/quantity.py]: There should be 2 lines between classes and functions. Please fix PEP8 warnings reported at https://travis-ci.org/astropy/astropy/jobs/348076730 (I just restarted the job because Travis has problem displaying the log).

File: astropy/units/quantity.py [hunk 5]
```diff
@@ -39,6 +40,15 @@
...
 _UFUNCS_FILTER_WARNINGS = {np.arcsin, np.arccos, np.arccosh, np.arctanh}
 
+def quantity_allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
+    """
+    Returns True if two arrays are element-wise equal within a tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
+    :func:`numpy.allclose`.
+    """
+    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
```
Comment (pllim) [inline astropy/units/quantity.py]: @mhvk , should `_unquantify_allclose_arguments` be moved over too? (Since this is a "hidden" function, I don't think deprecation would be necessary if we decide to move it.)

File: astropy/units/quantity.py [hunk 6]
```diff
@@ -687,6 +696,7 @@ def _result_as_quantity(self, result, unit, out):
         out._set_unit(unit)
         return out
 
+
```
Comment (pllim) [inline astropy/units/quantity.py]: Remove extra line.

File: astropy/utils/decorators.py [hunk 7]
```diff
@@ -115,7 +115,7 @@ def deprecated_func(*args, **kwargs):
                 category = AstropyDeprecationWarning
 
             warnings.warn(message, category, stacklevel=2)
-
+
```
Comment (pllim) [inline astropy/utils/decorators.py]: Please have your editor set to remove extraneous whitespace like this.

**Comment by rohanrajpal** [comment]
@mwcraig

File: astropy/units/quantity.py (hunk 5 above)
Comment (Cadair) [reply astropy/units/quantity.py]: Surely without moving that over `astropy.units.quantity` now has an import dependency on pytest?!

File: astropy/units/quantity.py [hunk 8]
```diff
@@ -39,6 +39,15 @@
 _UNIT_NOT_INITIALISED = "(Unit not initialised)"
 _UFUNCS_FILTER_WARNINGS = {np.arcsin, np.arccos, np.arccosh, np.arctanh}
 
+def quantity_allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
```
Comment (Cadair) [inline astropy/units/quantity.py]: I would vote for this just being called `allclose` as it's in `units.quantity`

File: CHANGES.rst [hunk 9]
```diff
@@ -61,6 +61,7 @@ astropy.table
 
 astropy.tests
 ^^^^^^^^^^^^^
+-Deprecated function ``quantity``allclose to helper
```
Comment (mhvk) [inline CHANGES.rst]: Formatting is wrong (see elsewhere in fine). Also, this is not very helpful to someone who is reading this. Write
```
The function ``quantity_allclose`` was moved to the ``units`` package with the new,
shorter name ``allclose``. It is deprecated in ``tests``. [#7252]
```

File: CHANGES.rst [hunk 10]
```diff
@@ -71,7 +72,7 @@ astropy.time
 
 astropy.units
 ^^^^^^^^^^^^^
-
+-Added function ``quantity``allclose to quantity
```
Comment (mhvk) [inline CHANGES.rst]: Again, formatting is wrong (need space and empty line after, plus link to PR), and the description is too terse to be useful:
```
Added a units-aware ``allclose`` function (this was previously available in the
``tests`` module as ``quantity_allclose``) [#7252]
```

File: astropy/tests/helper.py [hunk 11]
```diff
@@ -465,17 +464,16 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
                                                                rtol, atol),
                                **kwargs)
 
-
+@deprecated("3.1","moved function to astropy.units.quantity")
```
Comment (mhvk) [inline astropy/tests/helper.py]: This is not a helpful message. I think we can just use `alternative` instead of the `message` argument, as in `alternative='astropy.units.allclose'`

File: astropy/tests/helper.py [hunk 12]
```diff
@@ -465,17 +464,16 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
...
     """
     Returns True if two arrays are element-wise equal within a tolerance.
 
     This is a :class:`~astropy.units.Quantity`-aware version of
     :func:`numpy.allclose`.
     """
-    import numpy as np
-    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
-                       **kwargs)
+    from ..units.quantity import quantity_allclose as quantity_allclose_units
```
Comment (mhvk) [inline astropy/tests/helper.py]: I think we seem to have reached agreement to call it `allclose` in `units`

File: astropy/units/quantity.py (hunk 8 above)
Comment (mhvk) [reply astropy/units/quantity.py]: Agreed. It should also be added to `__all__`, so that it is exposed as `u.allclose()`.

Aside: needs an extra empty line above the function definition (pep8)

File: astropy/units/quantity.py [hunk 13]
```diff
@@ -39,6 +39,15 @@
...
+    """
+    Returns True if two arrays are element-wise equal within a tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
```
Comment (mhvk) [inline astropy/units/quantity.py]: Now that we are making this a much more visible function, it should have a proper docstring, defining the arguments, etc.  See developer documentation (and other examples in code) for details on how this looks. This particular sentence would go under "Notes"

File: astropy/units/quantity.py (hunk 5 above)
Comment (mhvk) [reply astropy/units/quantity.py]: Yes, `_unquantity_allclose_arguments` definitely needs to be moved here.

Comment (rohanrajpal) [reply astropy/units/quantity.py]: And the name `_unquantity_allclose_arguments ` shall remain the same or changed like `allclose` ?

File: astropy/units/quantity.py (hunk 8 above)
Comment (rohanrajpal) [reply astropy/units/quantity.py]: After going through `units.quantity` all the attributes of __all__ .All of them are classes which have functions inside them , so i have to do the same?

Comment (mhvk) [reply astropy/units/quantity.py]: No, this would remain a function - it is just a coincidence that in this module only classes are exported.

File: astropy/units/quantity.py (hunk 5 above)
Comment (mhvk) [reply astropy/units/quantity.py]: I think the name is fine as is -- and it is a private function anyway, so we can change it later if needed.

File: astropy/units/quantity.py (hunk 13 above)
Comment (rohanrajpal) [reply astropy/units/quantity.py]: That means `_unquantify_allclose_arguments` shall also have a proper docstring,right?

**Comment by rohanrajpal** [comment]
Okay @mhvk 
The changes you've requested have been done but the following remain
`
1. Change all occurrences of quantity_allclose in the code base (probably mostly in test files, but check).
2. Look for a good place in docs/units to mention the function (not in "getting started", probably).
`
I am manually trying to find the occurences but would appreciate if you could specify where to look, how to and which test files to modify.

File: astropy/units/quantity.py (hunk 13 above)
Comment (mhvk) [reply astropy/units/quantity.py]: It remains a private function and it is pretty obvious what it does, so I don't think it is essential, but feel free to add something short.

**Comment by mhvk** [comment]
@rohanrajpal - Just do the following:
```
find astropy -name \*.py -exec grep [^_]quantity_allclose {} \; -print
```
Note that it is fine to change not the tests themselves but just the import, as in:
```
from ...units import allclose as quantity_allclose
```

And the deprecation warnings will tell you whether you missed anything.

File: astropy/tests/helper.py [hunk 14]
```diff
@@ -465,48 +464,18 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
...
     :func:`numpy.allclose`.
     """
-    import numpy as np
-    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
-                       **kwargs)
-
+    from ..units.quantity import allclose as quantity_allclose_units
+    quantity_allclose_units(a, b, rtol=1.e-5, atol=None, **kwargs)
 
+@deprecated("3.1",alternative='astropy.units._unquantify_allclose_arguments')
```
Comment (mhvk) [inline astropy/tests/helper.py]: This one doesn't have to be deprecated - it is a private function. However, you do need to import it inside `assert_quantity_allclose`

File: CHANGES.rst (hunk 9 above)
Comment (mhvk) [reply CHANGES.rst]: The comments about the changelog should still be addressed.

File: astropy/units/quantity.py [hunk 15]
```diff
@@ -30,7 +30,7 @@
                               check_output)
 
 __all__ = ["Quantity", "SpecificTypeQuantity",
-           "QuantityInfoBase", "QuantityInfo"]
+           "QuantityInfoBase", "QuantityInfo","allclose"]
```
Comment (mhvk) [inline astropy/units/quantity.py]: Space after `,` - please follow pep8.

File: astropy/units/quantity.py [hunk 16]
```diff
@@ -40,6 +40,52 @@
 _UFUNCS_FILTER_WARNINGS = {np.arcsin, np.arccos, np.arccosh, np.arctanh}
 
 
+def allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
```
Comment (mhvk) [inline astropy/units/quantity.py]: This is a bit of a detail, but this function is not as important as the `Quantity` class - I would suggest moving it to the end of the file (with `_unquantify_allclose_arguments`)

File: astropy/units/quantity.py [hunk 17]
```diff
@@ -841,7 +887,7 @@ def to(self, unit, equivalencies=[]):
         --------
         to_value : get the numerical value in a given unit.
         """
-        # We don't use `to_value` below since we always want to make a copy
+        # We don't use `to_value` below since we always want to make a Copy
```
Comment (mhvk) [inline astropy/units/quantity.py]: Why this change? Just keep as is.

File: astropy/tests/helper.py (hunk 14 above)
Comment (rohanrajpal) [reply astropy/tests/helper.py]: Okay, but I do have to make a call to the new location of the function, in order to avoid code duplication.

**Comment by rohanrajpal** [comment]
I've tried to make all the changes following pep8, I've noticed the pep8 error 
`helper.py:482:80: E501 line too long (87 > 79 characters)`
in a lot of places, is this ignored while doing pep8 tests?

File: astropy/tests/helper.py (hunk 14 above)
Comment (mhvk) [reply astropy/tests/helper.py]: That why I noted that you have to import it in `assert_quantity_allclose`

**Comment by mhvk** [comment]
Indeed, long lines are ignored by default (though it is good to stick to it).

File: astropy/units/quantity.py (hunk 17 above)
Comment (rohanrajpal) [reply astropy/units/quantity.py]: My bad have fixed this in the recent pull request
@mhvk can you review it?

File: CHANGES.rst [hunk 18]
```diff
@@ -61,6 +61,10 @@ astropy.table
 
 astropy.tests
 ^^^^^^^^^^^^^
+- The function ``quantity_allclose`` was moved to the ``units`` package with the new,
+  shorter name ``allclose``. It is deprecated in ``tests``. [#7252]
+
+- The function ``_unquantify_allclose_arguments`` was moved to the ``units`` package. It is deprecated in ``tests``. [#7252]
```
Comment (mhvk) [inline CHANGES.rst]: This is a private function, so we should not advertise it in any way (i.e., just remove this from the changelog).

File: CHANGES.rst [hunk 19]
```diff
@@ -71,6 +75,11 @@ astropy.time
 
 astropy.units
 ^^^^^^^^^^^^^
+-  Added a units-aware ``allclose`` function (this was previously available in the
+  ``tests`` module as ``quantity_allclose``) [#7252]
+
+-  Added a units-aware ``_unquantify_allclose_arguments`` function (this was previously available in the
```
Comment (mhvk) [inline CHANGES.rst]: As above - remove this from the changelog.

File: astropy/coordinates/tests/test_api_ape5.py [hunk 20]
```diff
@@ -17,11 +17,11 @@
 from numpy.random import randn
 from numpy import testing as npt
 
-from ...tests.helper import (raises, quantity_allclose as allclose,
-                             assert_quantity_allclose as assert_allclose)
+from ...tests.helper import (raises, assert_quantity_allclose as assert_allclose)
```
Comment (mhvk) [inline astropy/coordinates/tests/test_api_ape5.py]: parentheses can go now...

File: astropy/coordinates/tests/test_celestial_transformations.py [hunk 21]
```diff
@@ -9,11 +9,10 @@
 from ..builtin_frames import (ICRS, FK5, FK4, FK4NoETerms, Galactic,
                               Supergalactic, Galactocentric, HCRS, GCRS, LSR)
 from .. import SkyCoord
-from ...tests.helper import (quantity_allclose as allclose,
-                             assert_quantity_allclose as assert_allclose)
+from ...tests.helper import (assert_quantity_allclose as assert_allclose)
```
Comment (mhvk) [inline astropy/coordinates/tests/test_celestial_transformations.py]: Remove parentheses

File: astropy/coordinates/tests/test_transformations.py [hunk 22]
```diff
@@ -11,8 +11,9 @@
 from .. import representation as r
 from ..baseframe import frame_transform_graph
 from ...tests.helper import (assert_quantity_allclose as assert_allclose,
-                             quantity_allclose, catch_warnings)
+                              catch_warnings)
```
Comment (mhvk) [inline astropy/coordinates/tests/test_transformations.py]: Not aligned properly

File: astropy/tests/helper.py [hunk 23]
```diff
@@ -3,15 +3,14 @@
...
-
 import os
 import sys
 import types
 import pickle
 import warnings
 import functools
-
 import pytest
+from ..utils.decorators import deprecated
```
Comment (mhvk) [inline astropy/tests/helper.py]: As I'm nitpicking anyway, an open line before this statement (separating standard library from our own code).

File: astropy/tests/helper.py [hunk 24]
```diff
@@ -461,52 +460,24 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
...
     :func:`numpy.allclose`.
     """
-    import numpy as np
-    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
-                       **kwargs)
+    from ..units.quantity import allclose as quantity_allclose_units
+    quantity_allclose_units(a, b, rtol=1.e-5, atol=None, **kwargs)
 
 
 def _unquantify_allclose_arguments(actual, desired, rtol, atol):
```
Comment (mhvk) [inline astropy/tests/helper.py]: This one should be removed altogether. Nobody should be relying on this private routine.

File: astropy/units/quantity.py (hunk 17 above)
Comment (mhvk) [reply astropy/units/quantity.py]: I still see a change here. Can you undo it?

File: astropy/units/quantity.py [hunk 25]
```diff
@@ -1722,3 +1722,50 @@ def _set_unit(self, unit):
...
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
+    :func:`numpy.allclose`.
+    """
+    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
+                       **kwargs)
+
+
+def _unquantify_allclose_arguments(actual, desired, rtol, atol):
+    from .. import units as u
```
Comment (mhvk) [inline astropy/units/quantity.py]: We are inside `units` here, so no need for this import; just change `u.Quantity` -> `Quantity` below (and same for `UnitsError`)

**Comment by mhvk** [comment]
Quite a few nitpicks, but also some more substantial comments. While you are at it, could you also rebase & squash the commits. And of course please check the test failures (much easier if you run tests locally first to ensure things are OK).

File: astropy/tests/helper.py (hunk 24 above)
Comment (rohanrajpal) [reply astropy/tests/helper.py]: You mean to remove the function `quantity_allclose` altogether?
Shouldn't a deprecated function call to the new location of the function?

Comment (mhvk) [reply astropy/tests/helper.py]: No, please look where the comment is made: you should remove `_unquantity_allclose_arguments` from `utils` -- it is a private function, so no deprecation is needed.

Comment (rohanrajpal) [reply astropy/tests/helper.py]: You meant to remove from `units` not `utils`, right? If you do mean units, isn't that the place I moved the function `_unquantity_allclose_arguments` to?Also, this comment is made in tests/helper.py.Should I remove it from `helper` as well or keep it as it is.

File: astropy/coordinates/tests/accuracy/test_ecliptic.py [hunk 26]
```diff
@@ -5,7 +5,7 @@
 
 import numpy as np
 
-from ....tests.helper import quantity_allclose
+from ....units.quantity import allclose as quantity_allclose
```
[the remark this answers is not in the document]
Comment (rohanrajpal) [reply astropy/coordinates/tests/accuracy/test_ecliptic.py]: Could you give a complete example? I'm getting a value error by importing with that method.Probably making a silly mistake.

**Comment by astrofrog** [comment]
I'm personally -1 on deprecating the old location (as in, I don't think we should raise a deprecation warning yet, but just silently allow it to work). This is a function that I've actually used in quite a few packages and I suspect others may have too, and it's one of these cases where it's not clear that deprecating it is actually necessary and is going to be a pain for affiliated package maintainers. I've been wanting for a while to write some kind of APE along these lines, but I don't think we always need to systematically emit deprecation warnings for old behavior (which then trigger CI failures) if keeping the old behavior isn't a burden (which it wouldn't be here).

File: astropy/tests/helper.py (hunk 24 above)
Comment (mhvk) [reply astropy/tests/helper.py]: I meant `utils` - you moved `_unquantity_allclose_arguments` to `units` and there is no reason to keep it here, as you import it directly from units in `assert_quantity_allclose`.

File: astropy/coordinates/tests/accuracy/test_ecliptic.py (hunk 26 above)
[the remark this answers is not in the document]
Comment (mhvk) [reply astropy/coordinates/tests/accuracy/test_ecliptic.py]:
```
from ....units import allclose as quantity_allclose
```

I hope this solves it; if not, please do debug yourself - this is taking me more time to answer than it should take you to solve.

**Comment by mhvk** [comment]
@astrofrog - OK, that makes sense to me as well. Just so we don't get another cycle
```
from ....units import allclose as quantity_allclose
```
and not define a deprecated `quantity_allclose` here.

**Comment by rohanrajpal** [comment]
Also thanks to all for help in this pull request :)

**Comment by Cadair** [comment]
Can this get a bug fix milestone? This is fixing a presumably unintended side effect of unbundling pytest in 3.0 I think.

**Comment by bsipocz** [comment]
cc @drdavella

**Comment by drdavella** [comment]
I am strongly in favor of the general direction of this PR. I have no objections to changing the milestone to a bugfix. I agree with @astrofrog that no external API changes or deprecations should be necessary (especially if we want it to be a bugfix).

File: astropy/tests/helper.py (hunk 23 above)
Comment (astrofrog) [reply astropy/tests/helper.py]: Remove this since it isn't used.

File: astropy/tests/helper.py [hunk 27]
```diff
@@ -461,6 +460,7 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
     :func:`numpy.testing.assert_allclose`.
     """
     import numpy as np
+    from ..units.quantity import _unquantify_allclose_arguments
     np.testing.assert_allclose(*_unquantify_allclose_arguments(actual, desired,
```
Comment (astrofrog) [inline astropy/tests/helper.py]: Instead of using a private function from astropy.units, why not just do:

```
assert quantity_allclose(...)
```

?

File: CHANGES.rst [hunk 28]
```diff
@@ -67,6 +67,11 @@ astropy.table
 astropy.tests
 ^^^^^^^^^^^^^
 
+- The function ``quantity_allclose`` was moved to the ``units`` package with the new,
+  shorter name ``allclose``. It is deprecated in ``tests``. [#7252]
```
Comment (astrofrog) [inline CHANGES.rst]: I think the last sentence should be something like 'The old import will continue to work but may be deprecated in future'

File: CHANGES.rst [hunk 29]
```diff
@@ -67,6 +67,11 @@ astropy.table
...
+
+- The function ``_unquantify_allclose_arguments`` was moved to the ``units`` package. It is deprecated in ``tests``. [#7252]
```
Comment (astrofrog) [inline CHANGES.rst]: Remove from changelog as it's private API.

File: astropy/tests/helper.py [hunk 30]
```diff
@@ -473,40 +473,5 @@ def quantity_allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
...
-
-    rtol = u.Quantity(rtol, subok=True, copy=False)
-    try:
-        rtol = rtol.to(u.dimensionless_unscaled)
-    except Exception:
-        raise u.UnitsError("`rtol` should be dimensionless")
-
-    return actual.value, desired.value, rtol.value, atol.value
+    from ..units.quantity import allclose as quantity_allclose_units
+    quantity_allclose_units(a, b, rtol=1.e-5, atol=None, **kwargs)
```
Comment (astrofrog) [inline astropy/tests/helper.py]: I'm confused, why is this called ``quantity_allclose_units`` and not just e.g. ``quantity_allclose``? (seems there is some redundancy).

File: astropy/tests/helper.py [hunk 31]
```diff
@@ -21,6 +19,7 @@
 except ImportError:
     pass
 
+from ...units.quantity import allclose as quantity_allclose
```
Comment (mhvk) [inline astropy/tests/helper.py]: Import from `...units` here in case we decide to move it (not unlikely)

File: astropy/tests/helper.py [hunk 32]
```diff
@@ -460,53 +459,4 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
...
-                               .format(atol.unit, actual.unit))
-
-    rtol = u.Quantity(rtol, subok=True, copy=False)
-    try:
-        rtol = rtol.to(u.dimensionless_unscaled)
-    except Exception:
-        raise u.UnitsError("`rtol` should be dimensionless")
-
-    return actual.value, desired.value, rtol.value, atol.value
+    assert quantity_allclose(actual, desired, rtol, atol, **kwargs)
```
Comment (mhvk) [inline astropy/tests/helper.py]: I was about to send the following comment on this suggestion by @astrofrog when the changes were pushed :-)... not sure I agree:
"""
Before we make that change: `assert_allclose` prints out some specifics about what is not matching, which `assert quantity_allclose` would not do.

One solution to importing a private function might be to not make it private (but not list it in `__all__` in `quantity.py` (or perhaps move it to `quantity_helper`).

But perhaps best would be to have a public helper function (or decorator?) a bit like `quantity_input` that allows one to pass in that the units of some arguments should be the consistent.
"""
For now, my suggestion would be to keep using `_unquantify_allclose_arguments`, and keep the wish for a new function separate -  See #7368.

File: astropy/units/quantity.py [hunk 33]
```diff
@@ -1722,3 +1722,61 @@ def _set_unit(self, unit):
                  ", so cannot set it to '{0}'.".format(unit)))
 
         super()._set_unit(unit)
+
+
+def isclose(a, b, rtol=1.e-5, atol=None, **kwargs):
```
Comment (mhvk) [inline astropy/units/quantity.py]: This now definitely is getting in the terrain of making more sense as a separate module/file! `quantity_helper` is an option, but perhaps better would be `quantity_utils`. But can be another PR.

File: docs/stats/lombscargle.rst [hunk 34]
```diff
@@ -407,7 +407,7 @@ Next we compute the two versions of the PSD from uniformly-sampled data:
 
 Examining the results, we see that the two outputs match:
 
->>> from astropy.tests.helper import quantity_allclose
+>>> from astropy.units.quantity import allclose as quantity_allclose
```
Comment (mhvk) [inline docs/stats/lombscargle.rst]: Again, import from `astropy.units` so we are not stuck with having it in `quantity.py`.

File: astropy/units/quantity.py (hunk 33 above)
Comment (pllim) [reply astropy/units/quantity.py]: Yeah, let's defer moving it to separate module in a separate PR.

**Comment by pllim** [comment]
@mhvk , I did a review and addressed your remaining concerns (missing doc and tests). Please have a look. If tests are passing now and you are satisfied, feel free to merge.

File: docs/units/quantity.rst [hunk 35]
```diff
@@ -97,6 +97,17 @@ If you want the value of the quantity in a different unit, you can use
...
+Comparing quantities
+====================
+
+|quantity| objects can be compared as follows:
+
+    >>> from astropy import units as u
+    >>> u.allclose([1, 2] * u.m, [100, 200] * u.cm)
+    True
+    >>> u.isclose([1, 2] * u.m, [100, 20] * u.cm)
+    array([ True, False])
```
Comment (mhvk) [inline docs/units/quantity.rst]: This is one of the annoying things that changed in recent numpy (and thus doesn't work across versions), that in numpy <=1.13, this has ", dtype=bool" appended; maybe just `doctest: +SKIP`...

File: astropy/tests/helper.py [hunk 36]
```diff
@@ -30,7 +29,7 @@
 
 __all__ = ['raises', 'enable_deprecations_as_exceptions', 'remote_data',
            'treat_deprecations_as_exceptions', 'catch_warnings',
-           'assert_follows_unicode_guidelines', 'quantity_allclose',
```
Comment (bsipocz) [inline astropy/tests/helper.py]: Can we do this on a bugfix branch? While importing with `*` is bad practice, this removal is an API change for those who does that.

**Comment by bsipocz** [comment]
As @astrofrog commented above (https://github.com/astropy/astropy/pull/7252#pullrequestreview-112022600), we need to remove this function for the ``__all__``, but that needs to be done in a follow-up PR milestoned to 3.1, so the bugfix branch keeps working without issue.
Whoever merges, please also open that simple PR to it doesn't get forgotten.

**Comment by pllim** [comment]
@bsipocz and @drdavella , adding it back in `__all__` breaks the Sphinx build in Travis. That is why I removed it in the first place. I am not comfortable merging this knowing Sphinx build would break.

**Comment by Cadair** [comment]
You could always just use this as the master fix and then use my other PR as the backport. Completely removes any API change in the bugfix then.

**Comment by Cadair** [comment]
(for @bsipocz we could merge that PR first and then revert the commit in this one) :wink:

**Comment by bsipocz** [comment]
@pllim - Yes, that's a good point, and I'm not comfortable with breaking sphinx either (but somehow missed that it happened already).

Probably the simplest is to add it back either during the backport (either as part of this or a separate commit), or in a PR directly opened to 2.0.x once this one is backported.

File: astropy/tests/helper.py [hunk 37]
```diff
@@ -459,12 +459,7 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
     This is a :class:`~astropy.units.Quantity`-aware version of
     :func:`numpy.testing.assert_allclose`.
     """
-    import numpy as np
-    from astropy.units.quantity import _unquantify_allclose_arguments
-
-    np.testing.assert_allclose(*_unquantify_allclose_arguments(actual, desired,
-                                                               rtol, atol),
-                               **kwargs)
+    assert _quantity_allclose(actual, desired, rtol=rtol, atol=atol, **kwargs)
```
Comment (pllim) [inline astropy/tests/helper.py]: @astrofrog , like this, correct?

**Comment by pllim** [comment]
:horse: :skull:

**Comment by pllim** [comment]
(I'll open a follow-up PR about that `__all__` thingy.)

**Comment by cdeil** [comment]
This PR broke some tests we have in Gammapy that call `assert_quantity_allclose`, they now fail with a non-informative `AssertionError` not saying if the value or unit is wrong:
https://travis-ci.org/gammapy/gammapy/jobs/367497442#L3786

Example:
```
>>> from astropy.tests.helper import assert_quantity_allclose
>>> import astropy.units as u
>>> assert_quantity_allclose(42 * u.m, 44 * u.m)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/tests/helper.py", line 466, in assert_quantity_allclose
    assert _quantity_allclose(actual, desired, rtol=rtol, atol=atol, **kwargs)
AssertionError
```

I will just remove all use of Quantity assert helper function calls to Astropy now from Gammapy.
Personally I prefer to have a separate assert on the unit anyways because the unit shouldn't just change unless we change something in our code.

But maybe it's possible to improve this to avoid this breaking change that might affect others?

File: astropy/tests/helper.py [hunk 38]
```diff
@@ -460,53 +459,10 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
     This is a :class:`~astropy.units.Quantity`-aware version of
     :func:`numpy.testing.assert_allclose`.
     """
-    import numpy as np
-    np.testing.assert_allclose(*_unquantify_allclose_arguments(actual, desired,
```
Comment (Cadair) [inline astropy/tests/helper.py]: It seems this line, which was reinstated because of a comment by @mhvk got deleted again in the sphinx fixes. @pllim can we put this back in your second PR, to fix @cdeil's problem?

**Comment by bsipocz** [comment]
Thanks @cdeil for catching this, I wish more packages were testing against the dev version to pick up issues like this so early.

File: astropy/tests/helper.py (hunk 38 above)
Comment (pllim) [reply astropy/tests/helper.py]: Ops. I only saw Tom's comment. Sorry. I'll fix this shortly.

**Comment by pllim** [comment]
Ah nevermind, it is the extra information... Ignore me. (Need more coffee)

**Comment by cdeil** [comment]
@pllim -  ☕️ ☕️ ☕️ ☕️ ☕️ ☕️ ☕️ ☕️
